src/librustc_mir/borrow_check/nll/mod.rs

```diff
@@ -202,11 +202,13 @@ fn dump_mir_results<'a, 'gcx, 'tcx>(
     });
 
     // Also dump the inference graph constraints as a graphviz file.
-    let _: io::Result<()> = do catch {
+    // `do catch` changed to IIFE for bootstrapping; consider changing back
+    // when we get a new stage0 compiler
+    let _: io::Result<()> = (||{
```

Comment:
How about we make a macro for this instead?

```
#[cfg(stage0)]
macro_rules do_catch {
  ($t:expr) = { (|| $t)() }
}

#[cfg(not(stage0))]
macro_rules do_catch {
  ($t:expr) = { do catch { $t } }
}
```

Reply:
And this way it can have ok-wrapping! :+1: